synthesizer/program/src/lib.rs

```diff
@@ -642,6 +648,14 @@ impl<N: Network, Instruction: InstructionTrait<N>, Command: CommandTrait<N>> Pro
         "type",
         "future",
     ];
+    /// A list of restricted keywords for Aleo programs, enforced at the VM-level for program hygiene.
+    /// Each entry is a tuple of the consensus version and a list of keywords.
+    /// If the current consensus version is greater than or equal to the specified version,
+    /// the keywords in the list should be restricted.
+    #[rustfmt::skip]
+    pub const RESTRICTED_KEYWORDS: &'static [(ConsensusVersion, &'static [&'static str])] = &[
+        (ConsensusVersion::V6, &["constructor"])
```

Comment:
Given that our transaction validity rules change from the V6 consensus height onwards, we need to clear the `partially_verified_transactions` cache.

Given that this is a one-time action (as opposed to program upgrades which can happen more frequently), you can just copy this approach: https://github.com/ProvableHQ/snarkVM/blob/mainnet/synthesizer/src/vm/mod.rs#L440

@raychu86 what do you think about just automatically clearing the entire cache on every consensus version upgrade so we don't have to think about it anymore. This could be e.g. sth like `if N::CONSENSUS_VERSION_HEIGHTS.iter().map(|_,h|h).contains(block.height()) { clear the cache }`.

Reply:
Just to double check my understanding, it seems like is only read once [here](https://github.com/ProvableHQ/snarkVM/blob/5d0d1745d05619e2d3aecdcf54cf65106e52f4b7/synthesizer/src/vm/verify.rs#L141).
If `is_partially_verified` is `true`, then an internal check in `check_fee` and `check_deployment_internal` are skipped. 
The check is added [here](https://github.com/ProvableHQ/snarkVM/blob/45c3ba4723d5fe0d119aec83dc6e3ac5abf5d324/synthesizer/src/vm/verify.rs#L167), so would that mean it's run every time, despite whether or not it is cached?

Reply:
You are right, I think I got this right the first time I reviewed but I forgot, thx for catching.